UPGRADE-GUIDE.md

```diff
@@ -0,0 +1,460 @@
+# Upgrade Guide
+
+This document provides a full guide for upgrading a Cosmos SDK chain from `v0.50.x` to `v0.53.x`.
```

Comment:
We should change this to something like "full guide for adding all new optional features ..."

Reply:
i added a line identifying the exact amounts of required and optional changes